needle-slack-connector/src/server/backend/connectors-backend.ts

```diff
@@ -0,0 +1,314 @@
+  deleteConnector,
+  publishConnectorRun,
+  type Session,
+  type ConnectorRunDescriptor,
+  createNeedleFileId,
+} from "@needle-ai/needle-sdk";
+import { type CanvasFileMetadata, slackConnectorsTable } from "../db/schema";
+import { db } from "../db";
+import { eq } from "drizzle-orm";
+import {
+  getConnectorDetails,
+  getCurrentFiles,
+  handleDatabaseUpdates,
+  updateLastSynced,
+  createSlackConnectorRecord,
+  deleteSlackConnectorRecords,
+  getSlackConnectorWithFiles,
+} from "./slack-db";
+import {
+  computeCanvasDiff,
+  createNewFiles,
+  processExistingFiles,
+} from "./dif-utils";
+import { createSlackService } from "../slack/service";
+import { type ExistingFile } from "../slack/types";
+
+export async function createSlackConnector(
+  params: CreateConnectorRequest,
+  session: Session,
+) {
+  if (!params.channels.length) {
+    throw new Error("At least one channel must be provided");
+  }
+
+  const connector = await createConnector(
+    {
+      name: params.name,
+      cronJob: params.cronJob,
+      cronJobTimezone: params.cronJobTimezone,
+      collectionIds: [params.collectionId],
+      credentials: params.credentials,
+    },
+    session.id,
+  );
+
+  try {
+    await createSlackConnectorRecord(
+      connector.id,
+      params.channels,
+      params.timezone,
+    );
+    await runSlackConnector({ connectorId: connector.id }, session);
+    return connector;
+  } catch (error) {
+    await deleteConnector(connector.id, session.id);
+    throw error;
+  }
+}
+
+export async function runSlackConnector(
+  { connectorId, simulateDate }: { connectorId: string; simulateDate?: Date },
+  session: Session,
+) {
+  const effectiveDate = simulateDate ?? new Date();
+
+  if (!session) {
+    throw new Error("Session required for connector run");
+  }
+
+  const [connector, connectorDetails] = await Promise.all([
+    getConnector(connectorId, session.id),
+    getConnectorDetails(connectorId),
+  ]);
+
+  if (!connector) throw new Error(`No connector found for ID: ${connectorId}`);
+  if (!connectorDetails)
+    throw new Error(`No Slack connector found for ID: ${connectorId}`);
+
+  const currentFiles = await getCurrentFiles(connectorId);
+
+  // Get current channel list from Slack to verify channel existence
+  const slackService = createSlackService(connector.credentials ?? "");
+  const channelsResponse = (await slackService.getChannels()) as {
+    ok: boolean;
+    channels?: { id: string; name: string }[];
+  };
+
+  if (!channelsResponse.ok || !channelsResponse.channels) {
+    throw new Error("Failed to fetch channels from Slack");
+  }
+
+  // Filter out deleted channels from connector's channelInfo
+  const existingChannelIds = new Set(
+    channelsResponse.channels.map((c) => c.id),
+  );
+  const updatedChannelInfo =
+    connectorDetails.channelInfo?.filter((channel) =>
+      existingChannelIds.has(channel.id),
+    ) ?? [];
+
+  // Update connector's channelInfo if channels were removed
```

Comment:
in this case i rather prefer just throwing and saying connector is broken. selected channels are a contract - if they break the contract then it's better that connector doesn't run anymore.